pkg/chunked/filesystem_linux_test.go

```diff
@@ -334,3 +338,49 @@ func createTempFile(t *testing.T, dir, name string) *os.File {
 	require.NoError(t, err)
 	return tmpFile
 }
+
+func TestSplitPath(t *testing.T) {
+	tests := []struct {
+		path         string
+		expectedDir  string
+		expectedBase string
+	}{
+		{"", "/", "."},
+		{".", "/", "."},
+		{"..", "/", "."},
+		{"../..", "/", "."},
+		{"../../..", "/", "."},
+		{"../../../foo", "/", "foo"},
+		{"../../../foo/..", "/", "."},
+		{"../../../foo/./../foo/bar/baz", "/foo/bar/", "baz"},
+		{"../../../foo/bar", "/foo/", "bar"},
```

Comment:
Is the trailing `/` in `dir` acceptable? With https://github.com/containers/storage/pull/2194#discussion_r1878843841 , it seems to me that it would interfere with the automatic creation of parent directories.

Reply:
that is a direct consequence of applying https://github.com/containers/storage/pull/2194#discussion_r1878704500

Added a  `strings.TrimSuffix` to drop any trailing slash, except for the root directory